### ISSUE TITLE
Accessibility / Customization Options

### ISSUE THREAD

**Issue by abuckles-uci** [usnistgov/NEMO#311]
Hi @rptmat57,

What are the plans / current processes for NEMO in regards to accessibility? This is in response to [Fact Sheet: New Rule on the Accessibility of Web Content and Mobile Apps Provided by State and Local Governments](https://www.ada.gov/resources/2024-03-08-web-rule/).

We've been using the [WAVE](https://wave.webaim.org/extension/) plugin for determining compliance so far. There are a range of issues that come up when using the plugin. Some have to do with color contrast issues and others with HTML structure (such as missing labels). I'm not sure how this all fits in with use of Bootstrap and FullCalendar libraries.

If you'd like I can send a more detailed list of possible issues I've found so far.

One more suggestion, it could be helpful to have some more color customization options.
1. Primary, secondary, warning, etc. button colors
2. Default calendar event color customization option where `#88B7CD` is: https://github.com/usnistgov/NEMO/blob/a06186bd10d6d27bdccc73a2d88dcbb21c140860/NEMO/templates/calendar/reservation_event_feed.html#L25
    1. This is an example of low color contrast

Thank you,
Aaron

**Comment by rptmat57** [comment]
Hi @abuckles-uci,
We do our best to adhere to the accessibility guidelines, but NEMO isn't 100% compliant. We use the Wave extension too.
I would gladly look at any PR aimed at increasing NEMO's compliance with accessibility.

The color customization are a bit tricky to add since we use bootstrap for a lot of those. Also, I gave up on making fullcalendar compatible, the version we use is too old and to hard to customize.

**Comment by abuckles-uci** [comment]
Hi @rptmat57,

Thanks for the response! I have just one followup question. I'm sure these would both be huge tasks, but are there any long term plans for using newer versions of Bootstrap and FullCalendar? For example, newer versions of Bootstrap support color modes / theming: https://getbootstrap.com/docs/5.3/customize/color-modes/.

Thanks,
Aaron

**Comment by rptmat57** [comment]
Oh I have been thinking about it, but there is no long term plan at the moment unfortunately.

This would be a pretty big task indeed. At least FullCalendar is limited to just one page, but bootstrap is used all over the place so that would take quite some time.

**Comment by gbaloch** [comment]
I would be more than happy to try tackling this problem beginning by migrating to the newest version of Bootstrap (5.3.x) and making the entire site fully ARIA-compliant from there. It might be best to build a bespoke calendar solution than attempt to migrate fullcalendar, or try to use something newer such as [schedule-x](https://github.com/schedule-x/schedule-x), which uses modern JavaScript web frameworks compared to jQuery. (Perhaps jQuery should be dropped for a modern view rendering engine too for a true progressive web app?)  I will try to experiment with this in the coming days/weeks,.
